### PR TITLE
#11  - Feature: create() and push() dataset

### PR DESCRIPTION
**Pull request by kmanaseryan**
Inherited from #18. 

Adding only those points which are new, for previous implementation see #18

- [x] Changed `create()` and `push()` so that the return dataset is already converted into frictionless
- [x] Added JSDoc comments for the above functions to make it more clear to understand
- [x] Tests are written for `push()`, `create(obj)` and `create(str)` use cases
- [x] Made some small refactoring in the tests. Ideally this would be better to make in a separate PR, but since we are in rush included it here. 

cc: @mariorodeghiero, @anuveyatsu

### CODE REVIEW

File: test/methods.test.js [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+}
+
+const client = new Client(
+  config.authToken,
+  config.organizationId,
+  config.datasetId,
+  config.api
+)
+
+test('push a metadata', async (t) => {
```
Comment (rufuspollock) [inline test/methods.test.js:25]: If we wanted to get fancy we should have a test where we push a dataset with existing name and get a 409 conflict etc but that's minor.

**Review by rufuspollock — APPROVED** [review]
Minor comment mainly about docstring - if you can push that improvement after merge that would be great!

File: test/methods.test.js (hunk 1 above)
Comment (kmanaseryan) [reply test/methods.test.js:25]: yea thought about this too, we can do this later

**Comment by kmanaseryan** [comment]
@rufuspollock done 6719452008b94d51799d96984a5f69a8552fcbff

**Pull request merged**